ZCRMiOS/Classes/ZCRMRestClient.swift

```diff
@@ -46,4 +54,97 @@ public class ZCRMRestClient
             completion( user, response, error )
         }
     }
-}
+    
+    public func getAllDashBoards(then Oncompletion:@escaping ArrayOfDashBoard)
+    {
+        // 200 is the maxNumber of records that can be retreived in an API Call
+        DashBoardAPIHandler().getAllDashBoards(FromPage:1, WithPerPageOf: 200)
+        {
+            (dashBoards,bulkAPIResponse,error)  in
+            Oncompletion(dashBoards,bulkAPIResponse,error)
+        }
+    }
+    
+    public func getAllDashboards(FromPage page:Int?,PerPage perPage:Int?,then
+        Oncompletion:@escaping ArrayOfDashBoard)
+        
+    {
+        let unwrappedPage = page ?? 1
+        let unwrappedPerPage = perPage ?? 200
+        
+        DashBoardAPIHandler().getAllDashBoards(FromPage: unwrappedPage,
+                                               WithPerPageOf: unwrappedPerPage)
+        {
+            (dashBoards,bulkAPIResponse,error) in
+            Oncompletion(dashBoards,bulkAPIResponse,error)
+        }
+        
+    }
+    
+    public func getDashBoardWith(ID:Int64,then OnCompletion:@escaping dashBoard)
+    {
+        DashBoardAPIHandler().getDashBoardWith(ID: ID)
+        {
+            (dashBoard,APIResponse,error) in
+            OnCompletion(dashBoard,APIResponse,error)
+        }
+        
+    }
+    
+    
+    public func getComponentWith(ID cmpID: Int64, FromDashBoardID dbID: Int64,
+                                 OnCompletion: @escaping dashBoardComponent)
+        
+    {
+        DashBoardAPIHandler().getComponentWith(ID: cmpID, FromDashBoardID: dbID)
+        {
+            (dashBoardComponent,APIResponse,error) in
+            OnCompletion(dashBoardComponent,APIResponse,error)
+        }
+        
+    }
+    
+    
+    public func refreshComponentWith(ID cmpID: Int64,InDashBoardID dbID: Int64,OnCompletion: @escaping refreshResponse)
+    {
+        
+        DashBoardAPIHandler().refreshComponentWith(ID: cmpID, InDashBoardID: dbID)
+        {
+            
+            (APIResponse, error) in
+            OnCompletion(APIResponse,error)
+        }
+        
+    }
+    
+    
+    public func refreshDashBoardWith(ID dbID: Int64, OnCompletion: @escaping refreshResponse )
+    {
+        
+        DashBoardAPIHandler().refreshDashBoardWith(ID: dbID)
+        {
+            
+            (APIResponse, error) in
+            OnCompletion(APIResponse,error)
+        }
+        
+    }
+    
+    
+    public func getDashBoardComponentColorThemes( OnCompletion: @escaping ArrayOfColorThemes )
```

Comment:
All methods should be moved to ZCRMAnalytics

Reply:
Should I create a NEW CLASS  "ZCRMAnalytics " ?

Reply:
Yes.